src/util/live_performance.js

```diff
@@ -0,0 +1,137 @@
+
+    // Code may be hosted on various endpoints: CDN, self-hosted,
+    // from unpkg... so this check doesn't include mapbox HTTP URL
+    if (url.includes('mapbox-gl.js')) return 'javascript';
+    if (url.includes('mapbox-gl.css')) return 'css';
+
+    if (isMapboxHTTPFontsURL(url)) return 'font';
+    if (isMapboxHTTPSpriteURL(url)) return 'sprite';
+    if (isMapboxHTTPStyleURL(url)) return 'style';
+    if (isMapboxHTTPTileJSONURL(url)) return 'tilejson';
+
+    return 'other';
+}
+
+function getStyle(resourceTimers: Array<PerformanceEntry>): ?string {
+    if (resourceTimers) {
+        for (const timer of resourceTimers) {
+            const url = timer.name.split('?')[0];
+            if (isMapboxHTTPStyleURL(url)) {
+                const split = url.split('/').slice(-2);
+                if (split.length === 2) {
+                    return `${split[0]}:${split[1]}`;
+                }
+            }
+        }
+    }
+}
+
+export function getLivePerformanceMetrics(data: LivePerformanceData): LivePerformanceMetrics {
+    const resourceTimers = window.performance.getEntriesByType('resource');
+    const resourcesByType = categorize(resourceTimers, getResourceCategory);
+    const counters = getTransferRangePerResourceType(resourcesByType);
+    const devicePixelRatio = window.devicePixelRatio;
+    const connection = window.navigator.connection || window.navigator.mozConnection || window.navigator.webkitConnection;
+    const metrics = {counters: [], metadata: [], attributes: []};
+
+    const addMetric = (arr, name, value) => {
+        if (value) {
+            arr.push({name, value: value.toString()});
+        }
+    };
+
+    for (const counter in counters) {
+        addMetric(metrics.counters, counter, counters[counter]);
+    }
+    if (data.interactionRange[0] !== +Infinity && data.interactionRange[1] !== -Infinity) {
+        addMetric(metrics.counters, "interactionRangeMin", data.interactionRange[0]);
+        addMetric(metrics.counters, "interactionRangeMax", data.interactionRange[1]);
+    }
+
+    addMetric(metrics.attributes, "style", getStyle(resourceTimers));
+    addMetric(metrics.attributes, "terrainEnabled", data.terrainEnabled ? "true" : "false");
+    addMetric(metrics.attributes, "fogEnabled", data.fogEnabled ? "true" : "false");
+    addMetric(metrics.attributes, "projection", data.projection.name);
+
+    addMetric(metrics.metadata, "devicePixelRatio", devicePixelRatio);
+    addMetric(metrics.metadata, "connectionEffectiveType", connection ? connection.effectiveType : undefined);
+    addMetric(metrics.metadata, "cpuCores", window.navigator.hardwareConcurrency);
+    addMetric(metrics.metadata, "userAgent", window.navigator.userAgent);
+    addMetric(metrics.metadata, "screenWidth", window.screen.width * devicePixelRatio);
+    addMetric(metrics.metadata, "screenHeight", window.screen.height * devicePixelRatio);
+    addMetric(metrics.metadata, "windowWidth", window.innerWidth * devicePixelRatio);
```

Comment:
I'd suggest not multiplying `devicePixelRatio` into this. It keeps the definition closer to what window.innerWidth` is. Also, the logical pixel size would be more useful for normalizing things like loading performance than the physical pixel size.

Reply:
No strong opinion on that one, we can always retrieve the physical or logical size since we store the device pixel ratio anyway. I'll update for logical size.